File: core/src/main/java/com/emarsys/core/concurrency/CoreSdkHandlerProvider.java
package com.emarsys.core.concurrency;

import android.os.Handler;
import android.os.HandlerThread;

import java.util.UUID;

public class CoreSdkHandlerProvider {

    public Handler provideHandler() {
        HandlerThread handlerThread = new HandlerThread("CoreSDKHandlerThread-" + UUID.randomUUID().toString());
        handlerThread.start();
        return new CoreSdkHandler(handlerThread);
    }

}

File: core/src/main/java/com/emarsys/core/concurrency/CoreSdkHandlerProvider.kt
package com.emarsys.core.concurrency

import android.os.HandlerThread
import com.emarsys.core.handler.CoreSdkHandler
import java.util.*

class CoreSdkHandlerProvider {
    fun provideHandler(): CoreSdkHandler {
        val handlerThread = HandlerThread("CoreSDKHandlerThread-" + UUID.randomUUID().toString())
        handlerThread.start()
        return CoreSdkHandler(CoreHandler(handlerThread))
    }
}